### PR TITLE
Add possibilty to hide fields in diagram

### PR DESCRIPTION
**Pull request by johannesloher**

### CODE REVIEW

File: src/diagramDrawing.ts [hunk 1]
```diff
@@ -1,6 +1,29 @@
 import { appendFileSync, createWriteStream, writeFileSync } from 'fs';
 import { generate } from 'node-plantuml';
 import { OUTPUT_DIR } from '.';
+import { isArrayOfStrings } from './util';
```
Comment (sken) [inline src/diagramDrawing.ts]: maybe other functions are also suited to be moved to this file
* objectFromEntries
* trim
?

File: src/diagramDrawing.ts [hunk 2]
```diff
@@ -10,8 +33,74 @@ function getOutputFile(scenario: string): string {
...
-function extractPayload(dict: unknown): string {
-    return JSON.stringify(dict, null, 1);
+function objectFromEntries(arr: [string, unknown][]): object {
+    return Object.assign({}, ...Array.from(arr, ([k, v]) => ({ [k]: v })));
+}
+
+function hideFields(payload: object, hiddenFields: string[]): object {
+    return objectFromEntries(
+        Object.entries(payload).map(([key, value]) =>
+            hiddenFields.includes(key) ? [key, '***'] : [key, value],
```
Comment (sken) [inline src/diagramDrawing.ts]: move '***' to a const

File: src/diagramDrawing.ts [hunk 3]
```diff
@@ -10,8 +33,74 @@ function getOutputFile(scenario: string): string {
...
+function hidePlaintextIfNeeded(payload: string, hidePlaintext = false): string {
+    return hidePlaintext ? '***' : payload;
+}
+
+function formatBinaryPayload(payload: Buffer): string {
+    return `binary data (${(payload as Buffer).length} bytes)`;
+}
+
+function trim(text: string, length: number): string {
+    return text.length > length ? `${text.substring(0, length - 3)}...` : text;
```
Comment (sken) [inline src/diagramDrawing.ts]: Why exactly 3 ?

Comment (johannesloher) [reply src/diagramDrawing.ts]: `'...'`.length === 3`

File: src/diagramDrawing.ts [hunk 4]
```diff
@@ -10,8 +33,74 @@ function getOutputFile(scenario: string): string {
     return `${OUTPUT_DIR()}/_${scenario}.png`;
 }
 
-function extractPayload(dict: unknown): string {
-    return JSON.stringify(dict, null, 1);
+function objectFromEntries(arr: [string, unknown][]): object {
+    return Object.assign({}, ...Array.from(arr, ([k, v]) => ({ [k]: v })));
+}
```
Comment (sken) [inline src/diagramDrawing.ts]: why not the spread operator?

File: src/diagramDrawing.ts (hunk 1 above)
Comment (johannesloher) [reply src/diagramDrawing.ts]: done

File: src/diagramDrawing.ts (hunk 4 above)
Comment (johannesloher) [reply src/diagramDrawing.ts]: doesn't work

File: src/diagramDrawing.ts (hunk 2 above)
Comment (johannesloher) [reply src/diagramDrawing.ts]: done

**Pull request merged**